hack/generator/pkg/codegen/storage/storage_type_factory.go

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package storage
+
+import (
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astmodel"
+	"github.com/pkg/errors"
+	"k8s.io/klog/v2"
+)
+
+// StorageTypeFactory is used to create storage types for a specific service
+type StorageTypeFactory struct {
+	service                    string                                                  // Name of the group we're handling (used mostly for logging)
```

Comment:
copying my comment from previous review over: why not call this `group`, since that's what it's called elsewhere in the code? I feel like using `service` terminology here is adding additional terminology overhead. People already need to understand what a `group` and `version` are (to understand what storage conversions even do in the context of Kuberentes)

Reply:
Thanks, I thought I'd fixed all of those.